terraform/modules/marklogic/variables.tf

```diff
@@ -100,6 +100,10 @@ variable "dap_external_role_arns" {
   type = list(string)
 }
 
+variable "dap_external_canonical_users" {
```

Comment:
You could have made this optional with a default of `[]` so then test/production don't need to pass `[]` into the module.